cmd/sink/k8s/objects.go

```diff
@@ -0,0 +1,23 @@
+// Copyright KubeArchive Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package k8s
+
+import (
+	"time"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+// GetLastUpdateTs returns the last time obj was updated on the cluster
+func GetLastUpdateTs(obj *unstructured.Unstructured) time.Time {
+	fields := obj.GetManagedFields()
+	var latest *metav1.Time
```

Comment:
This looks like it can be `nil` since its a pointer, if we are sure this won't be `nil` under any case its fine, if not we should do something with it. Maybe a default value.

Reply:
I don't _think_ that we can get a resource where the `metadata.managedFields` list is empty, but to be safe I'm testing if I can set it to the zero value, which is Jan 1, year 1, 00:00:00.000000000 UTC. This way latest can't be nil

Reply:
setting latest to the zero value for `metav1.Time` does not break the logic AND guarantees that `lastest` won't be `nil`